src/libcollections/string.rs

```diff
@@ -671,6 +671,29 @@ impl String {
     pub fn clear(&mut self) {
         self.vec.clear()
     }
+
+    /// The format function takes a precompiled format string and a list of
+    /// arguments, to return the resulting formatted string.
+    ///
+    /// # Arguments
+    ///
+    ///   * args - a structure of arguments generated via the `format_args!` macro.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// let s = String::format(format_args!("Hello, {}!", "world"));
+    /// assert_eq!(s, "Hello, world!".to_string());
+    /// ```
+    #[unstable(feature = "collections",
+               reason = "duplicates std::fmt::format, only needed when libstd is missing")]
+    pub fn format(args: fmt::Arguments) -> String {
+        // FIXME #21826
+        use core::fmt::Writer;
+        let mut output = String::new();
+        let _ = write!(&mut output, "{}", args);
+        output
+    }
```

Comment:
Yes, I think String::format is a better, more descriptive name.
On Feb 4, 2015 6:46 AM, "Keegan McAllister" <notifications@github.com>
wrote:

> In src/libcollections/string.rs
> <https://github.com/rust-lang/rust/pull/21912#discussion_r24047873>:
>
> > +    ///
> > +    /// # Example
> > +    ///
> > +    /// ```rust
> > +    /// let s = String::format(format_args!("Hello, {}!", "world"));
> > +    /// assert_eq!(s, "Hello, world!".to_string());
> > +    /// ```
> > +    #[unstable(feature = "collections",
> > +               reason = "duplicates std::fmt::format, only needed when libstd is missing")]
> > +    pub fn format(args: fmt::Arguments) -> String {
> > +        // FIXME #21826
> > +        use core::fmt::Writer;
> > +        let mut output = String::new();
> > +        let _ = write!(&mut output, "{}", args);
> > +        output
> > +    }
>
> I think String::format is a better, more descriptive name. Is there a
> specific reason we can't revisit this API choice?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/21912/files#r24047873>.
>